src/RangeParser.php

```diff
@@ -0,0 +1,185 @@
+    {
+        if ($value === null || $value === 'empty') {
+            return null;
+        }
+
+        if (!preg_match('/^(?P<open>\[|\()(?P<lower>[^,]*),(?P<upper>[^\)\]]*)(?P<close>\)|\])$/', $value, $matches)) {
+            throw new InvalidArgumentException('Unsupported range format');
+        }
+
+        $lower = $matches['lower'] ? trim($matches['lower'], '"') : null;
+        $upper = $matches['upper'] ? trim($matches['upper'], '"') : null;
+        $includeLower = $matches['open'] === '[';
+        $includeUpper = $matches['close'] === ']';
+
+        if ($lower === null && $upper === null) {
+            return [null, null];
+        }
+
+        return match($this->type) {
+            Schema::TYPE_INT_RANGE => self::parseIntRange($lower, $upper, $includeLower, $includeUpper),
+            Schema::TYPE_BIGINT_RANGE => self::parseBigIntRange($lower, $upper, $includeLower, $includeUpper),
+            Schema::TYPE_NUM_RANGE => self::parseNumRange($lower, $upper),
+            Schema::TYPE_DATE_RANGE => self::parseDateRange($lower, $upper, $includeLower, $includeUpper),
+            Schema::TYPE_TS_RANGE => self::parseTsRange($lower, $upper),
+            Schema::TYPE_TS_TZ_RANGE => self::parseTsTzRange($lower, $upper),
+            default => [$lower, $upper]
+        };
+    }
+
+    private static function parseIntRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
+    {
+        $min = $lower === null ? null : (int) $lower;
+        $max = $upper === null ? null : (int) $upper;
+
+        if ($min !== null && $includeLower === false) {
+            $min += 1;
+        }
+
+        if ($max !== null && $includeUpper === false) {
+            $max -= 1;
+        }
+
+        return [$min, $max];
+    }
+
+    private static function parseBigIntRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
+    {
+        if (PHP_INT_SIZE === 8) {
+            return self::parseIntRange($lower, $upper, $includeLower, $includeUpper);
+        }
+
+        [$min, $max] = self::parseNumRange($lower, $upper);
+
+        if ($min !== null && $includeLower === false) {
+            /** @var float $min */
+            $min += 1;
+        }
+
+        if ($max !== null && $includeUpper === false) {
+            /** @var float $max */
+            $max -= 1;
+        }
+
+        return [$min, $max];
+    }
+
+    private static function parseNumRange(?string $lower, ?string $upper): array
+    {
+        $min = $lower === null ? null : (float) $lower;
+        $max = $upper === null ? null : (float) $upper;
+
+        return [$min, $max];
+    }
+
+    private static function parseDateRange(?string $lower, ?string $upper, bool $includeLower, bool $includeUpper): array
+    {
+        $interval = new DateInterval('P1D');
+        $min = $lower ? DateTime::createFromFormat('Y-m-d', $lower) : null;
+        $max = $upper ? DateTime::createFromFormat('Y-m-d', $upper) : null;
```

Comment:
Need to decide this will be `DateTime` or better `DateTimeImmutable` 
or just string due to `yiisoft/db` do not support `DateTime` object. 
And `DateTimeImmutable` can be added after solving the issue yiisoft/db#725 perhaps in version 2.0.0